cg/meta/upload/scout/uploadscoutapi.py

```diff
@@ -455,3 +378,72 @@ def get_config_builder(self, analysis, hk_version) -> ScoutConfigBuilder:
         }
 
         return config_builders[analysis.pipeline]
+
+    def create_rna_dna_mapper(self, rna_case: models.Family) -> Dict[str, Dict[str, list]]:
+        """Build nested relationship dictionary for DNA samples and their DNA cases based on each RNA sample
+        subject_id in a case. This allows for upload of all RNA samples in an RNA case to all matching DNA samples in
+        multiple DNA cases. Example dict row: {rna_sample_id : {dna_sample_id : [dna_case_1, dna_case_2]}}. For a
+        trio of RNA samples matching a trio of DNA samples, it would have three of these rows.
+
+        Args:
+            rna_case     (models.Family):           Case
+        Returns:
+            rna_to_dna_mapper     (Dict):       rna-dna relationships, and corresponding dna cases based on subject id
+        """
+        rna_to_dna_mapper = {}
+        for link in rna_case.links:
+            rna_sample = link.sample
+            if not rna_sample.subject_id:
+                raise CgDataError(
+                    f"Failed on RNA sample {rna_sample.internal_id} as subject_id field is empty"
+                )
+
+            rna_to_dna_mapper[rna_sample.internal_id] = {}
+            matching_samples = self.status_db.samples_by_subject_id(
+                customer_id=rna_case.customer.internal_id,
+                subject_id=rna_sample.subject_id,
+                is_tumour=rna_sample.is_tumour,
+            )
+            if len(matching_samples.all()) > 2:
+                raise CgDataError(
+                    "Failed to upload files for RNA case: multiple DNA samples linked to it via subject_id"
+                )
+
+            sample: models.Sample
+            for sample in matching_samples:
+                if sample.internal_id == rna_sample.internal_id:
+                    continue
+                rna_to_dna_mapper[rna_sample.internal_id][sample.name] = []
+                for link in sample.links:
+                    case_object: models.Family = link.family
+                    if (
+                        case_object.data_analysis in [Pipeline.MIP_DNA, Pipeline.BALSAMIC]
+                        and case_object.customer_id == rna_case.customer_id
+                    ):
+                        rna_to_dna_mapper[rna_sample.internal_id][sample.name].append(
+                            case_object.internal_id
+                        )
+        rna_to_dna_mapper_error = self.rna_to_dna_mapper_correct(
+            rna_to_dna_mapper=rna_to_dna_mapper
+        )
+        if not rna_to_dna_mapper_error:
+            raise CgDataError()
+
+        return rna_to_dna_mapper
+
+    def rna_to_dna_mapper_correct(self, rna_to_dna_mapper: Dict[str, Dict[str, list]]) -> bool:
+        """While executing create_rna_dna_mapper: Check whether no DNA samples were found based on subject_id of the
```

Comment:
I think `"""Check whether no DNA samples were found based on subject_id of the RNA samples."""` is enough

Reply:
Thanks, added to next commit.